telegram/ext/_applicationbuilder.py

```diff
@@ -885,6 +899,32 @@ def updater(self: BuilderType, updater: Optional[Updater]) -> BuilderType:
         self._updater = updater
         return self
 
+    def post_init(
+        self: BuilderType, post_init: Callable[[Application], Coroutine[Any, Any, None]]
+    ) -> BuilderType:
+        """
+        Sets a callback to be executed by :meth:`Application.run_polling` and
+        :meth:`Application.run_webhook` *after* executing :meth:`Application.initialize` but
+        *before* executing :meth:`Updater.start_polling` or :meth:`Updater.start_webhook`,
+        respectively.
+
+        Tip:
+            This can be used for custom startup logic that requires to await coroutines, e.g.
+            setting up the bots commands via :meth:`~telegram.Bot.set_my_commands`.
+
```

Comment:
Maybe worth including an example?
```suggestion
            setting up the bots commands via :meth:`~telegram.Bot.set_my_commands`.
            
            Example::
            
                async def callback(app: Application) -> None:
                    ...
                
                application = Application.builder().token("TOKEN").post_init(callback).build()

```